enterprise/views.py

```diff
@@ -416,3 +419,121 @@ def post(self, request):
+                the request querystring.
+
+        """
+        try:
+            client = CourseApiClient()
+            course_details = client.get_course_details(course_id)
+        except HttpClientError:
+            raise Http404
+
+        if course_details is None:
+            raise Http404
+
+        enterprise_customer_uuid = request.GET.get('ec_uuid')
+        if not enterprise_customer_uuid:
+            raise Http404
+
+        try:
+            enterprise_customer_uuid = UUID(enterprise_customer_uuid)
+            # pylint: disable=no-member
+            enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+        except (ValueError, EnterpriseCustomer.DoesNotExist):
+            raise Http404
+
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
+        course_name = course_details['name']
+        course_image_uri = course_details['media']['course_image']['uri']
+        course_organization = course_details['org']
+        course_short_description = course_details['short_description']
+        course_pacing = course_details['pacing'].title()
+        course_start_date = ''
+        if course_details['start']:
+            course_start_date = parse(course_details['start']).strftime('%B %d, %Y')
+
+        context_data = {
+            'page_title': _('Choose Your Track'),
+            'page_language': get_language_from_request(request),
+            'platform_name': platform_name,
+            'course_id': course_id,
+            'course_name': course_name,
+            'course_organization': course_organization,
+            'course_short_description': course_short_description,
+            'course_pacing': course_pacing,
+            'course_start_date': course_start_date,
+            'course_image_uri': course_image_uri,
+            'enterprise_customer': enterprise_customer,
+            'enterprise_welcome_text': self.enterprise_welcome_text.format(
+                enterprise_customer_name=enterprise_customer.name,
+                platform_name=platform_name,
+                strong_start='<strong>',
+                strong_end='</strong>',
+            ),
+            'confirmation_text': self.confirmation_text,
+            'starts_at_text': self.starts_at_text,
+            'course_pace_text': self.course_pace_text,
+            'view_course_details_text': self.view_course_details_text,
+            'select_mode_text': self.select_mode_text,
+            'enroll_text': self.enroll_text,
+            'audit_text': self.audit_text,
+            'price_text': self.price_text,
+            'free_price_text': self.free_price_text,
+            'discount_text': self.discount_text.format(
+                enterprise_customer_name=enterprise_customer.name,
+            ),
+            'not_eligible_for_certificate_text': self.not_eligible_for_cert_text,
+            'continue_link_text': self.continue_link_text,
+        }
```

Comment:
@douglashall I agree that for now most of the content is just static string but the current implementation gives us flexibility. Here are my thoughts on this:
* Its easy to test the text content from the response context.
* Future manipulation of text (like adding dynamic values in the strings) can be done directly in the view without doing changes to the html. HTML gets hard to read when we add too much logic into the template.
* Easy to use different templating language without any extra effort.
* Consistent with the existing approach for the enterprise views
I agree that for now most of the content is just static string but this will give us flexibility. Please let me know your thoughts.